dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -870,7 +912,17 @@ public WorkflowAction findAction(final String actionId,
 									 final User user) throws DotDataException, DotSecurityException {
 
 		Logger.debug(this, "Finding the action: " + actionId + " for the step: " + stepId);
-		return this.workFlowFactory.findAction(actionId, stepId);
+		final WorkflowAction workflowAction = this.workFlowFactory.findAction(actionId, stepId);
+		if (null != workflowAction) {
+			if (!SYSTEM_WORKFLOW_ID.equals(workflowAction.getSchemeId())) {

```

Comment:
This could be a method, assertValidLicense (final String schemeId) {

if (!SYSTEM_WORKFLOW_ID.equals(workflowAction.getSchemeId()) && !hasValidLicense() && !this.getFriendClass().isFriend()) {

     throw ...